utils/config.go

```diff
@@ -533,6 +533,13 @@ func GenerateClientConfig(c *model.Config, diagnosticId string, license *model.L
 	props["EnableTutorial"] = strconv.FormatBool(*c.ServiceSettings.EnableTutorial)
 	props["ExperimentalEnableDefaultChannelLeaveJoinMessages"] = strconv.FormatBool(*c.ServiceSettings.ExperimentalEnableDefaultChannelLeaveJoinMessages)
 	props["ExperimentalGroupUnreadChannels"] = *c.ServiceSettings.ExperimentalGroupUnreadChannels
+
+	if *c.ServiceSettings.ExperimentalChannelOrganization || *c.ServiceSettings.ExperimentalGroupUnreadChannels != model.GROUP_UNREAD_CHANNELS_DISABLED {
```

Comment:
Is this extra check of `ExperimentalGroupUnreadChannels` needed when we already look at its value when setting a default? This seems to make the new setting redundant since it could just be keyed off `ExperimentalGroupUnreadChannels`, but maybe I'm missing something

Reply:
Setting to default has a different purpose than this check.

Once the config has been retrieved, we are moving existing users from `ExperimentalGroupUnreadChannels` over to `ExperimentalOrgnaization`.

This was the solution @esethna proposed in order to move existing users properly over to the new Channel Organization settings.

Not a common pattern but it's not often MM moves from one experimental flag to another

Reply:
@hmhealey we want to make sure there is no change for end users. ie if the group unreads setting is enabled, and the user has it on we want to migrate them to these new channel org settings without changing their sidebar behavior unexpectedly.

If there is a better way to accomplish [this expected behaviour](https://github.com/mattermost/mattermost-webapp/pull/1374#issuecomment-400757689) then we can update.

Reply:
So will the old setting have no effect then? That's where I'm getting a bit lost. Or does it turn on if either setting is used?

As these are experimental settings, I'm much less concerned with breaking the existing behaviour

Reply:
> As these are experimental settings, I'm much less concerned with breaking the existing behaviour

If this is the case it would simplify the code. 

Like @esethna mentioned I added the change to ensure there are no changes for the end user, this was based on feedback Eric provided in the webapp PR.


So just as you stated, the new sidebar settings are turned on if either setting is used.